src/commands/new.ts

```diff
@@ -387,13 +388,71 @@ export default {
     // #region Run Packager Install
     // pnpm/yarn/npm install it
 
+    // check if there is a cache of node_modules using a hash of the package.json
+    function hash(str: string) {
+      return crypto.createHash("md5").update(str).digest("hex")
+    }
+
+    // check if there is a cache of node_modules using a hash of the package.json
+    const packageJsonHash = hash(packageJsonRaw)
+
+    // store hash at `~/.ignite/cache/${packageJsonHash}`
+    const cachePath = path(filesystem.homedir(), ".ignite", "cache", packageJsonHash)
+    const cacheExists = exists(cachePath) === "dir"
+
+    // construct cache paths
+    const lockFile = {
+      yarn: "yarn.lock",
+      pnpm: "pnpm-lock.yaml",
+      npm: "package-lock.json",
+    }
+    const cache = {
+      node_modules: path(cachePath, "node_modules"),
+      lock: path(cachePath, lockFile[packagerName]),
+      pods: path(cachePath, "ios", "Pods"),
+      build: path(cachePath, "ios", "build"),
+    }
+    const target = {
+      node_modules: path(targetPath, "node_modules"),
+      lock: path(targetPath, lockFile[packagerName]),
+      pods: path(targetPath, "ios", "Pods"),
+      build: path(targetPath, "ios", "build"),
+    }
+
+    // if there is a cache, copy it over to the target path node_modules
+    if (cacheExists) {
+      startSpinner("Copying cached node_modules")
+
+      // ensure that target path node_modules exists
+      filesystem.dir(target.node_modules)
+      // copy the cache to the target path node_modules
+      copy(cache.node_modules, target.node_modules, { overwrite: true })
+      copy(cache.lock, target.lock, { overwrite: true })
+      copy(cache.pods, target.pods, { overwrite: true })
+      copy(cache.build, target.build, { overwrite: true })
+      stopSpinner("Copying cached node_modules", "📦")
+    }
+
     if (installDeps === true) {
       const unboxingMessage = `Unboxing ${packagerName} dependencies`
       startSpinner(unboxingMessage)
       await packager.install({ ...packagerOptions, onProgress: log })
       stopSpinner(unboxingMessage, "🧶")
     }
 
+    // if there is no cache, create one
+    if (!cacheExists) {
```

Comment:
We'll want a flag to disallow this.

Reply:
Other ideas:

```
ignite-cli deps-cache # help
ignite-cli deps-cache path # returns location of cache
ignite-cli deps-cache clear
```

Reply:
I think `--deps-cache=false` would be a good option to add here. 

Logging the location of the cache path when `--debug` is enabled would be another good way to help users figure out what is going on